storey/steps/flatten.py

```diff
@@ -0,0 +1,21 @@
+from itertools import chain
+
+from storey import Flow, Event
+from storey.dtypes import _termination_obj
+
+
+class Flatten(Flow):
+    """
+    Flattens sequences of sequences (i.e lists of lists, sets of sets, etc...) into a single sequence, by default
+    casts sequence to list. When setting to_set=True, casts sequence to set.
```

Comment:
Afaict this does not flatten lists of lists or any other data structure, but rather splits an event with an iterable body into multiple events. I'm not sure which behaviour is intended - the one documented or the one implemented.